uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java

```diff
@@ -302,4 +306,82 @@ public void aggregateMarketplaceRating() {
+    public Set<IMarketplaceRating> getAllRatingsForPortletInLastXDays(
+            int numberOfDaysBack, final String fname) {
+
+        LocalDateTime todayMinusNumberOfDaysBack = LocalDateTime.now().minusDays(numberOfDaysBack);
+
+        final TypedQuery<MarketplaceRatingImpl> query =
+                this.createQuery(
+                        this.createCriteriaQuery(
+                                new Function<
+                                        CriteriaBuilder, CriteriaQuery<MarketplaceRatingImpl>>() {
+                                    @Override
+                                    public CriteriaQuery<MarketplaceRatingImpl> apply(
+                                            CriteriaBuilder input) {
+                                        final CriteriaQuery<MarketplaceRatingImpl> criteriaQuery =
+                                                input.createQuery(MarketplaceRatingImpl.class);
+                                        final Root<MarketplaceRatingImpl> definitionRoot =
+                                                criteriaQuery.from(MarketplaceRatingImpl.class);
+                                        Predicate ratingDatePredicate =
+                                                input.greaterThanOrEqualTo(
+                                                        definitionRoot.<java.util.Date>get(
+                                                                "ratingDate"),
+                                                        todayMinusNumberOfDaysBack.toDate());
+                                        Predicate conditionPortlet =
+                                                input.equal(
+                                                        definitionRoot
+                                                                .get("marketplaceRatingPK")
+                                                                .get("portletDefinition"),
+                                                        portletDefinitionDao
+                                                                .getPortletDefinitionByFname(
+                                                                        fname));
+                                        Predicate allConditions =
+                                                input.and(conditionPortlet, ratingDatePredicate);
+                                        criteriaQuery.select(definitionRoot).where(allConditions);
+
+                                        return criteriaQuery;
+                                    }
+                                }));
+
+        return new HashSet<IMarketplaceRating>(query.getResultList());
+    }
+
+    @Override
+    @PortalTransactionalReadOnly
+    @OpenEntityManager(unitName = PERSISTENCE_UNIT_NAME)
+    public Set<IMarketplaceRating> getAllRatingsInLastXDays(int numberOfDaysBack) {
+
+        final Calendar queryDate = Calendar.getInstance();
+        queryDate.add(Calendar.DAY_OF_YEAR, 0 - numberOfDaysBack);
```

Comment:
:question: Could the Java 8 Time processing utils be used to simplify/clarify the code?
http://www.baeldung.com/migrating-to-java-8-date-time-api

continued from https://github.com/Jasig/uPortal/pull/1064#discussion_r155367622

Reply:
made this change

Reply:
Thanks @phillips1021!Ccould the change also be pushed to GitHub?

Reply:
It was pushed to GitHub previously - see: https://github.com/phillips1021/uPortal/commit/3ec0bb9557ee2e7c4c08d525f66036768741a413

Reply:
line 358 uses the same subtract operation and has not been updated.